sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JsonInferSchema.scala

```diff
@@ -122,7 +122,7 @@ private[sql] class JsonInferSchema(options: JSONOptions) extends Serializable {
 
       case VALUE_STRING =>
         val field = parser.getText
-        val decimalTry = allCatch opt {
+        lazy val decimalTry = allCatch opt {
```

Comment:
+1, looks correct.

cc @HyukjinKwon , @MaxGekk , @cloud-fan

Reply:
Yup. This is what I suggested. I manually tested it as well. LGTM.

Reply:
We can. But looks using lazy already restores previous performance as was though.